src/ClientBuilder.php

```diff
@@ -26,14 +26,14 @@ final class ClientBuilder implements ClientBuilderInterface
     private $options;
 
     /**
-     * @var TransportFactoryInterface|null The transport factory
+     * @var TransportInterface The transport
      */
-    private $transportFactory;
+    private $transport;
 
     /**
-     * @var TransportInterface|null The transport
+     * @var HttpClientInterface The HTTP client
```

Comment:
```php
$httpClient = new class() implements HttpClientInterface {
    public function sendRequest(string $requestData, Options $options): Response
    {
        $client = HttpClient::create([
            'headers' => [
                'Content-Type' => 'application/x-sentry-envelope',
                'User-Agent' => Client::SDK_IDENTIFIER . '/' . Client::SDK_VERSION,
            ],
        ]);

        $response = $client->request('POST', $options->getDsn()->getEnvelopeApiEndpointUrl(), [
            'body' => $requestData,
        ]);

        return new Response($response->getStatusCode(), $response->getHeaders(), '');
    }
};

Sentry\init([
    'dsn' => env('SENTRY_DSN'),
    'http_client' => $httpClient,
]);
```
I like the UX of this approach much more than the boilerplate required today. https://docs.sentry.io/platforms/php/configuration/transports/

We could even go the extra mile and add two sample implementations for Symfony and Guzzle to the SDK.

Reply:
If it becomes super easy for SDK users to setup the Symfony or Guzzle HTTP client (without boilerplate) then this could be a net improvement :)  I do share concerns that there will be bugs (as well as missing features) when implementing your own HTTP client - we spent a lot of time working on HTTP client issues in the 1.x branch (although to be fair that branch was trying to do more than the 3.x branch, like sending multiple events in parallel at the end of the request).